### DISCUSSION TITLE
KiCAD conversion

### DISCUSSION THREAD

**Discussion by jhogendorn** [CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter#144] [General]
Hi, I've done a KiCAD 7 conversion of this project, mainly to spin an add-on board for myself due to the stock and shipping issues for my country (Australia).

I've confirmed the add-on board works, if there's interest I'm happy to provide back the KiCAD project files.

I probably won't pursue it now my needs are met but there's some scope there as well to do some basic conversions to make the board cheaper to spin on jlcpcb. primarily just through dropping the resistor arrays for standard 0603 resistors, changing out the 1/4w resistors for something in the jlc basic parts list (and maybe going to 0804). and swapping out some of the 0.33 caps. I can't remember but the crystal I think is an extended part as well, not sure if there's a basic equivalent. Oh and maybe switching the odd set 2mm pitch jumper headers for just basic 2.54 ones.

Considering each extended part is $3 to swap the reel it adds up fast. board cost could drop to < $30ea by doing this.

Anyway, if there's utility in the files I'm happy to provide.

**Comment by ICPete** [comment]
Hi Joshua,
I'd definitely be interested. I'm planning to install a system with several add-on boards (probably 4), plus the main board.
I've spent some time studying the schematics and Gerber files provided, but since I use KiCAD mostly, that would be dynamite if you could upload them. I might be able to pursue those part substitutions to reduce the cost at JLCPCB, as well.
Regards,
Pete

**Comment by jhogendorn** [comment]
[Energy-Meter-Kicad 2.zip](https://github.com/CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter/files/12076674/Energy-Meter-Kicad.2.zip)

Here's a zip of the files. Some thoughts:

- I am not offering any support, guarantee, assuming any liability etc. Assume a MIT license.
- I've only focussed on the add-on board, the main/remote boards, what may be there, I've not really done anything with. I know the add-on board is viable if you spin with jlcpcb but nothing else.
- the bom jlc inferres is iffy. I've tried to do some lcsc mappings but I'm not saying they're correct or the best choices.
- My changes from the upstream are largely just DRC violation fixes, clearance changes, moving silkscreens etc.

**Reply by ICPete** [reply]
Thank you Joshua,
I will download the files and take a look over the next week or two.

**Comment by PandemiK911** [comment]
Hey !
Using the Kicad files, I've been able to produce a STEP file for the add-on board. You can use it to create your own case design and check everything is ok before printing. Everything component on the card is named and can be hidden/disabled. Really useful for to stack them and show only the ESP32 on top add-on.

That makes a really big file and can't be uploaded to Github (25MB limit). You can find it here : https://www.swisstransfer.com/d/fc36a219-0388-48e9-a0ab-d1abcb7a1437

I've not idea how long it will stay available. If someone has any idea of how it could be down-sized, you're welcome !

Julien

**Reply by jhogendorn** [reply]
Would it not be best to upload it to printables or something?

**Reply by PandemiK911** [reply]
Great idea, thanks !
Done here : https://www.printables.com/model/722941-6chan-add-on-board

Is that ok for you to be credited as like this ? (first time I upload a model to printables)

**Reply by jhogendorn** [reply]
looks great :)
